### PR TITLE
Refactor transaction code

### PR DESCRIPTION
**Pull request by wangwenyue**
Abstract transaction detail component.

Delete too much duplicate code.

Merge two different transactions component into common one.

Make code more like React code.

Modify some typo and indent problem.

### CODE REVIEW

File: src/component/Transactions/txItem/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    componentDidMount() {
+        setInterval(() => {
+            window.web3.eth.getBlockNumber((err, res) => this.setState({
+                currentEthBlockNum: res
+            }))
+        }, 500)
+    }
+
+    parseTimeStamp = (timestamp) => {
```
Comment (Keith-CY) [inline src/component/Transactions/txItem/index.js]: use template string

File: src/component/Transactions/txItem/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        let date = new Date(timestamp)
+        let Y = date.getFullYear() + '-'
+        let M = (date.getMonth() + 1 < 10 ? '0' + (date.getMonth() + 1) : date.getMonth() + 1) + '-'
+        let D = date.getDate() + ' '
+        let h = (date.getHours() < 10 ? '0' + date.getHours() : date.getHours()) + ':'
+        let m = (date.getMinutes() < 10 ? '0' + date.getMinutes() : date.getMinutes())
+        return Y + M + D + h + m
+    }
+
+    parseValue = (value) => {
```
Comment (Keith-CY) [inline src/component/Transactions/txItem/index.js]: should be outside the class

File: src/component/Transactions/txItem/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        return Math.floor(value / 1e14) / 10000
+    }
+
+    render() {
+        const {
+            eth_tx_hash,
```
Comment (Keith-CY) [inline src/component/Transactions/txItem/index.js]: use camelCase

File: src/component/Transactions/txItem/index.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            ac_tx_hash,
+            value,
+            startedTime,
+            status,
+            eth_block_num,
+            txType,
+            wd_tx_hash,
+        } = this.props
+
+        const transactionType = {
```
Comment (Keith-CY) [inline src/component/Transactions/txItem/index.js]: should be outside the class

File: src/component/Transactions/txStatusBar/index.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+const imgPath = './Rectangle6.png'
+const imgPathDisabled = './Rectangle6_disable.png'
+
+class TxStatusBar extends React.Component {
+
+    isTxHashExist = () => {
+        return this.props.ac_tx_hash || this.props.eth_tx_hash
+    }
+
+    selectStage = (txType) => {
```
Comment (Keith-CY) [inline src/component/Transactions/txStatusBar/index.js]: should be outside the class

File: src/component/Transactions/txDetail/index.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        let qLink = [
+            this.queryLink(this.props.eth_tx_hash, 'eth_tx_hash'),
+            this.queryLink(this.props.ac_tx_hash || this.props.wd_tx_hash, 'ac_tx_hash')
+        ]
+
+        if(txType === 'eth2ebc') {
+            return [qLink[0], qLink[1]]
+        } else if(txType === 'ebc2eth') {
+            return [qLink[1], qLink[0]]
+        }
```
[the remark this answers is not in the document]
Comment (wangwenyue) [reply src/component/Transactions/txDetail/index.js]: No else branch, actually.
Solution is I rewrite this part of code.

**Comment by wangwenyue** [comment]
@Keith-CY Resolved all request changes.

**Pull request merged**